File: src/main/java/linkboard/data/dao/UserAccountDao.java
package linkboard.data.dao;

import linkboard.data.connection.ConnectionManager;
import linkboard.data.entity.UserAccountEntity;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

public class UserAccountDao
{
    public UserAccountEntity findByUsernameAndPassword(String username, String password)
    {
        try {
            Connection conn = ConnectionManager.getConnection();

            Statement stmt = conn.createStatement();
            ResultSet rs = stmt.executeQuery(String.format(
                    "SELECT * FROM user_account WHERE username = '%s' and password = '%s'",
                    username, password));

            return this.mapResultToUserAccount(rs);
        }
        catch (Exception e) {
            // TODO: Add logging
            throw new RuntimeException(e);
        }
    }

    public UserAccountEntity save(UserAccountEntity entity)
    {
        try {
            Connection conn = ConnectionManager.getConnection();

            Statement stmt = conn.createStatement();
            stmt.executeUpdate(String.format("INSERT INTO user_account " +
                                                     "(username, password, email, first_name, last_name) " +
                                                     "VALUES " +
                                                     "('%s', '%s', '%s', '%s', '%s')",
                                             entity.getUsername(),
                                             entity.getPassword(),
                                             entity.getEmail(),
                                             entity.getFirstName(),
                                             entity.getLastName())
                               );

            return entity;
        }
        catch (Exception e) {
            // TODO: Add logging
            throw new RuntimeException(e);
        }
    }

    private UserAccountEntity mapResultToUserAccount(ResultSet rs) throws SQLException
    {
        UserAccountEntity user = null;

        if (rs.next()) {
            user = new UserAccountEntity();

            user.setId(rs.getLong("id"));
            user.setUsername(rs.getString("username"));
            user.setPassword(rs.getString("password"));
            user.setEmail(rs.getString("email"));
            user.setFirstName(rs.getString("first_name"));
            user.setLastName(rs.getString("last_name"));
        }

        return user;
    }
}

File: node-express/node/data/dao/user-account-dao.js
var baseDao = require('./base-dao.js');
var userAccountMapper = require('../mapping/user-account-mapping.js');

module.exports.findByUsernameAndPassword = function(username, password, callback)
{
    var qry = 'SELECT * FROM user_account WHERE username = $1 AND password = $2';
    var params = [username, password];

    baseDao.executeQuery(qry, params, function(result) {
        if (result) {
            if (callback) {
                var bean = userAccountMapper.fromEntity(result.rows[0]);
                callback(bean);
            }
        }
    });
};

module.exports.save = function(user, callback)
{
    var qry = 'INSERT INTO user_account (username, password, email, first_name, last_name) VALUES ($1, $2, $3, $4, $5)';
    var params = [user.username, user.password, user.email, user.firstName, user.lastName];

    baseDao.executeQuery(qry, params, function(result) {
        if (result) {
            if (callback) {
                callback(result);
            }
        }
    });
};